test/dynamo/test_lazy_constant.py

```diff
@@ -0,0 +1,276 @@
+        eager_not_kw = fn(tensor_input, "foo")
+        compiled_not_kw = opt_fn(tensor_input, "foo")
+        self.assertTrue(same(eager_not_kw, compiled_not_kw))
+
+        # Both should cause recompilation since they branch differently
+        self.assertGreater(counter.frame_count, 1)
+
+    def test_keyword_iskeyword_no_branch(self):
+        """Test keyword.iskeyword without branching on result."""
+
+        def fn(x, word):
+            result = keyword.iskeyword(word)
+            return x + 1, result
+
+        counter = CompileCounter()
+        opt_fn = torch.compile(fn, backend=counter)
+
+        tensor_input = torch.randn(3)
+
+        eager0 = fn(tensor_input, "class")
+        compiled0 = opt_fn(tensor_input, "class")
+        self.assertEqual(eager0[1], compiled0[1])
+        self.assertTrue(same(eager0[0], compiled0[0]))
+
+        eager1 = fn(tensor_input, "notakeyword")
+        compiled1 = opt_fn(tensor_input, "notakeyword")
+        self.assertEqual(eager1[1], compiled1[1])
+        self.assertTrue(same(eager1[0], compiled1[0]))
+
+    def test_frozenset_contains(self):
+        """Test that frozenset.__contains__ is properly traced."""
+        valid_options = frozenset({"alpha", "beta", "gamma"})
+
+        def fn(x, option):
+            if option in valid_options:
+                return x + 1
+            return x - 1
+
+        counter = CompileCounter()
+        opt_fn = torch.compile(fn, backend=counter)
+
+        tensor_input = torch.randn(3)
+
+        # "alpha" is in the frozenset
+        eager_in = fn(tensor_input, "alpha")
+        compiled_in = opt_fn(tensor_input, "alpha")
+        self.assertTrue(same(eager_in, compiled_in))
+
+        # "delta" is not in the frozenset
+        eager_not_in = fn(tensor_input, "delta")
+        compiled_not_in = opt_fn(tensor_input, "delta")
+        self.assertTrue(same(eager_not_in, compiled_not_in))
+
+        # Both should cause recompilation since they branch differently
+        self.assertGreater(counter.frame_count, 1)
+
+    def test_frozenset_contains_no_branch(self):
+        """Test frozenset.__contains__ without branching on result."""
+        valid_options = frozenset({1, 2, 3})
+
+        def fn(x, value):
+            result = value in valid_options
+            return x + 1, result
+
+        counter = CompileCounter()
+        opt_fn = torch.compile(fn, backend=counter)
+
+        tensor_input = torch.randn(3)
+
+        eager0 = fn(tensor_input, 2)
+        compiled0 = opt_fn(tensor_input, 2)
+        self.assertEqual(eager0[1], compiled0[1])
+        self.assertTrue(same(eager0[0], compiled0[0]))
+
+        eager1 = fn(tensor_input, 5)
+        compiled1 = opt_fn(tensor_input, 5)
+        self.assertEqual(eager1[1], compiled1[1])
+        self.assertTrue(same(eager1[0], compiled1[0]))
+
+    def test_global_constant_passthrough(self):
+        """Test that global constants also benefit from lazy constant optimization."""
```

Comment:
This test should be modifying the global/closure cell and checking for no recompiles?